shopwild-frontend/src/reducer/data-reducer.js

```diff
@@ -0,0 +1,108 @@
+export const dataReducer = (state, action) => {
+  switch (action.type) {
+    case "INITIALIZE_DATA":
+      return {
+        ...state,
+        [action.payload.name]: action.payload.data,
+      };
+
+    case "MARK_WISHLIST_PRODUCTS":
+      return {
+        ...state,
+        allProducts: state.allProducts.map((product) =>
+          state.wishlist.find(
+            (item) => item.product.id === product.id && item.isWishlisted
+          )
+            ? { ...product, isWishlisted: true }
+            : product
+        ),
+      };
+
+    case "MARK_CART_PRODUCTS":
+      return {
+        ...state,
+        allProducts: state.allProducts.map((product) =>
+          state.cart.find(
+            (item) => item.product.id === product.id && item.quantity > 0
+          )
+            ? { ...product, isAddedToCart: true }
+            : product
+        ),
```

Comment:
try moving the functionalities into a seperate pure functions 
It will decrease the file length and will be better to interpret